### PR TITLE
Updated with latest APRL recommendations.

### PR DESCRIPTION
**Pull request by jimays-avila**
# Overview/Summary

ARG queries Output Standardization for Automation

[AB#31152](https://dev.azure.com/CSUSolEng/12ba7e46-a92a-47f6-ad6a-fba0479234a7/_workitems/edit/31152)
[AB#31316](https://dev.azure.com/CSUSolEng/12ba7e46-a92a-47f6-ad6a-fba0479234a7/_workitems/edit/31316)
[AB#31317](https://dev.azure.com/CSUSolEng/12ba7e46-a92a-47f6-ad6a-fba0479234a7/_workitems/edit/31317)
[AB#31318](https://dev.azure.com/CSUSolEng/12ba7e46-a92a-47f6-ad6a-fba0479234a7/_workitems/edit/31318)
[AB#31319](https://dev.azure.com/CSUSolEng/12ba7e46-a92a-47f6-ad6a-fba0479234a7/_workitems/edit/31319)
[AB#31320](https://dev.azure.com/CSUSolEng/12ba7e46-a92a-47f6-ad6a-fba0479234a7/_workitems/edit/31320)
[AB#30178](https://dev.azure.com/CSUSolEng/12ba7e46-a92a-47f6-ad6a-fba0479234a7/_workitems/edit/30178)

## This PR fixes/adds/changes/removes

1. Changed all kql queries to match standardization for automation

## As part of this Pull Request I have

- [ ] Read the [Contribution Guide](https://azure.github.io/Azure-Proactive-Resiliency-Library/contributing) and ensured this PR is compliant with the guide
- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/Azure-Proactive-Resiliency-Library/pulls)
- [x] Associated it with relevant [GitHub Issues](https://github.com/Azure/Azure-Proactive-Resiliency-Library/issues) or ADO Work Items (Internal Only)
- [ ] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Azure-Proactive-Resiliency-Library/tree/main)
- [ ] Ensured PR tests are passing
- [ ] Updated relevant and associated documentation (e.g. Contribution Guide, Docs etc.)

### CODE REVIEW

File: docs/content/services/networking/Application Gateway/code/appgw-9/appgw-9.kql
Comment (rodrigosantosms) [inline docs/content/services/networking/Application Gateway/code/appgw-9/appgw-9.kql:1]: add the following as a comment instead TBD.

// under development

File: docs/content/services/networking/Application Gateway/code/appgw-8/appgw-8.kql
Comment (rodrigosantosms) [inline docs/content/services/networking/Application Gateway/code/appgw-8/appgw-8.kql:1]: Query results should follow the following standard
"recommendationId", Name, Id
or
"recommendationId", Name, Id, AnyAdditionalParameters_if_Any

See, VMs, Storage Accounts, Load Balancers for some examples. The project will look like this:

| project recommendationId = "vm-1", name, id

File: docs/content/services/networking/Application Gateway/code/appgw-7/appgw-7.kql
Comment (rodrigosantosms) [inline docs/content/services/networking/Application Gateway/code/appgw-7/appgw-7.kql:1]: Query results should follow the following standard
"recommendationId", Name, Id
or
"recommendationId", Name, Id, AnyAdditionalParameters_if_Any

See, VMs, Storage Accounts, Load Balancers for some examples. The project will look like this:

| project recommendationId = "vm-1", name, id

File: docs/content/services/networking/Application Gateway/code/appgw-6/appgw-6.kql
Comment (rodrigosantosms) [inline docs/content/services/networking/Application Gateway/code/appgw-6/appgw-6.kql:1]: Query results should follow the following standard
"recommendationId", Name, Id
or
"recommendationId", Name, Id, AnyAdditionalParameters_if_Any

See, VMs, Storage Accounts, Load Balancers for some examples. The project will look like this:

| project recommendationId = "vm-1", name, id

File: docs/content/services/networking/Application Gateway/code/appgw-4/appgw-4.kql
Comment (rodrigosantosms) [inline docs/content/services/networking/Application Gateway/code/appgw-4/appgw-4.kql:1]: Query results should follow the following standard
"recommendationId", Name, Id
or
"recommendationId", Name, Id, AnyAdditionalParameters_if_Any

See, VMs, Storage Accounts, Load Balancers for some examples. The project will look like this:

| project recommendationId = "vm-1", name, id

File: docs/content/services/networking/Application Gateway/code/appgw-3/appgw-3.kql
Comment (rodrigosantosms) [inline docs/content/services/networking/Application Gateway/code/appgw-3/appgw-3.kql:1]: Query results should follow the following standard
"recommendationId", Name, Id
or
"recommendationId", Name, Id, AnyAdditionalParameters_if_Any

See, VMs, Storage Accounts, Load Balancers for some examples. The project will look like this:

| project recommendationId = "vm-1", name, id

File: docs/content/services/networking/Application Gateway/code/appgw-2/appgw-2.kql
Comment (rodrigosantosms) [inline docs/content/services/networking/Application Gateway/code/appgw-2/appgw-2.kql:1]: Query results should follow the following standard
"recommendationId", Name, Id
or
"recommendationId", Name, Id, AnyAdditionalParameters_if_Any

See, VMs, Storage Accounts, Load Balancers for some examples. The project will look like this:

| project recommendationId = "vm-1", name, id

File: docs/content/services/networking/Application Gateway/code/appgw-1/appgw-1.kql
Comment (rodrigosantosms) [inline docs/content/services/networking/Application Gateway/code/appgw-1/appgw-1.kql:1]: Query results should follow the following standard
"recommendationId", Name, Id
or
"recommendationId", Name, Id, AnyAdditionalParameters_if_Any

See, VMs, Storage Accounts, Load Balancers for some examples. The project will look like this:

| project recommendationId = "vm-1", name, id

File: docs/content/services/networking/Application Gateway/code/appgw-5/appgw-5.kql
Comment (rodrigosantosms) [inline docs/content/services/networking/Application Gateway/code/appgw-5/appgw-5.kql:1]: add the following as a comment instead TBD.

// under development

File: docs/content/services/networking/Application Gateway/_index.md [hunk 1]
```diff
@@ -208,11 +209,31 @@ Plan for backend maintenance by using connection draining. Connection draining h
...
+Application Gateway (Standard_v2 or WAF_v2 SKU) can support up to 125 instances. Although a /24 subnet isn't required per Application Gateway v2 SKU deployment, it is highly recommended. This is to ensure that Application Gateway v2 has sufficient space for autoscaling expansion and maintenance upgrades.
+
+**Resources**
+
+- [Azure Application Gateway infrastructure configuration | Microsoft Learn]([https://learn.microsoft.com/azure/application-gateway/application-gateway-autoscaling-zone-redundant](https://learn.microsoft.com/en-us/azure/application-gateway/configuration-infrastructure#size-of-the-subnet))
+
+**Resource Graph Query/Scripts**
+
+{{< collapse title="Show/Hide Query/Script" >}}
+{{< code lang="sql" file="code/appgw-1/appgw-9.kql" >}} {{< /code >}}
```
Comment (rodrigosantosms) [inline docs/content/services/networking/Application Gateway/_index.md:233]: Folder name is incorrect, it should be appgw-9

File: docs/content/services/networking/Application Gateway/_index.md [hunk 2]
```diff
@@ -208,11 +209,31 @@ Plan for backend maintenance by using connection draining. Connection draining h
...
 
 **Resource Graph Query/Scripts**
-
 {{< collapse title="Show/Hide Query/Script" >}}
 
 {{< code lang="sql" file="code/appgw-8/appgw-8.kql" >}} {{< /code >}}
 
 {{< /collapse >}}
 
+### APPGW-9 - Ensure Application Gateway Subnet is using a 24 subnet mask
```
Comment (rodrigosantosms) [inline docs/content/services/networking/Application Gateway/_index.md:218]: Add the / before 24.

### APPGW-9 - Ensure Application Gateway Subnet is using a /24 subnet mask

File: docs/content/services/networking/Application Gateway/_index.md [hunk 3]
```diff
@@ -18,10 +18,11 @@ The presented resiliency recommendations in this guidance include Application Ga
...
-| [APPGW-5 - Monitor and Log the configurations and traffic](#appgw-5---monitor-and-log-the-configurations-and-traffic)                       |  Medium  | Preview  | Yes |
+| [APPGW-5 - Monitor and Log the configurations and traffic](#appgw-5---monitor-and-log-the-configurations-and-traffic)                       |  Medium  | Preview  | No |
 | [APPGW-6 - Use Health Probes to detect backend availability](#appgw-6---use-health-probes-to-detect-backend-availability)                   |  Medium  | Preview  | Yes |
 | [APPGW-7 - Deploy backends in a zone-redundant configuration](#appgw-7---deploy-backends-in-a-zone-redundant-configuration)                 |  High    | Preview  | Yes |
 | [APPGW-8 - Plan for backend maintenance by using connection draining](#appgw-8---plan-for-backend-maintenance-by-using-connection-draining) |  Medium  | Preview  | Yes |
+| [APPGW-9 - Ensure Application Gateway Subnet is using a 24 subnet mask](#appgw-9---ensure-application-gateway-subnet-is-using-a-24-subnet-mask)                       |  High    | Preview  | Yes |
```
Comment (rodrigosantosms) [inline docs/content/services/networking/Application Gateway/_index.md:25]: Add the / before 24.

### APPGW-9 - Ensure Application Gateway Subnet is using a /24 subnet mask

**Review by rodrigosantosms — CHANGES_REQUESTED** [review]
PLease see the required changes for each file/line.

File: docs/content/services/networking/Application Gateway/code/appgw-9/appgw-9.kql
Comment (jimays-avila) [reply docs/content/services/networking/Application Gateway/code/appgw-9/appgw-9.kql:1]: I have updated all queries to match the recommendations as requested.

**Comment by jimays-avila** [comment]
@rodrigosantosms can you please review the changes.

**Comment by oZakari** [comment]
Hey @jimays-avila, you have some merge conflicts due to the [changes](https://github.com/Azure/Azure-Proactive-Resiliency-Library/tree/main/docs/content/services/networking/application-gateway/code) that were made for standardizing the service folder names and using CAF abbreviations. Once you correct that, let me know and I can review and merge into main for you,

**Comment by jimays-avila** [comment]
@oZakari I recently made changes based on those factors. Are they not reflecting as I made those updates yesterday.  Can this PR just be cancelled and I can submit a new one once all the changes have been made or will the new changes updated here as well?

**Comment by oZakari** [comment]
@jimays-avila  They are not reflecting in your current branch atm. Below is your current branch:

image

Here is what the current upstream main branch looks like:
image

A standard git merge should detect the renamed files, but there are some changes within the index.md that you may need to go through on a case-by-case basis. There are a bunch of tools to help with this if you want to avoid the command line and my favorite is https://www.gitkraken.com/

However, if you want to mess with it too much. You could create a new branch from upstream main and then just copy your changes in this time since it looks like the KQL queries were the only things modified. 

I'm happy to help as well, so feel free to ping me in Teams.

**Comment by jimays-avila** [comment]
I made the necessary changes to align with the naming convention of the folders and files and kql queries.

**Comment by oZakari** [comment]
Currently working with @jimays-avila on resolving merge conflicts.

File: docs/content/services/networking/application-gateway/_index.md
Comment (oZakari) [inline docs/content/services/networking/application-gateway/_index.md:1]: Please change these back to AGW acronym

File: docs/content/services/compute/virtual-machine-scale-sets/code/vmss-10/vmss-10.kql [hunk 4]
```diff
@@ -0,0 +1,7 @@
+// Azure Resource Graph Query
+// This query will check if the VMSS are currently using the latest image.  If not the Image reference will be empty
+resources
+| where type == "microsoft.compute/virtualmachinescalesets"
+| extend VMSSName = name
+| extend ImageReference = tostring(properties.virtualMachineProfile.storageProfile.imageReference.version)
+| project VMSSName, ImageReference
```
Comment (oZakari) [inline docs/content/services/compute/virtual-machine-scale-sets/code/vmss-10/vmss-10.kql:7]: Please add the standard query columns: id, name and us param1 in place of imageReference.

File: docs/content/services/networking/application-gateway/code/agw-8/agw-8.kql
Comment (oZakari) [inline docs/content/services/networking/application-gateway/code/agw-8/agw-8.kql:1]: Please add the standard query columns: id, name, etc

File: docs/content/services/networking/application-gateway/code/agw-7/agw-7.kql
Comment (oZakari) [inline docs/content/services/networking/application-gateway/code/agw-7/agw-7.kql:1]: Please add the standard query columns: id, name, etc

**Review by oZakari — CHANGES_REQUESTED** [review]

File: docs/content/services/compute/virtual-machine-scale-sets/code/vmss-10/vmss-10.kql (hunk 4 above)
Comment (jimays-avila) [reply docs/content/services/compute/virtual-machine-scale-sets/code/vmss-10/vmss-10.kql:7]: Will make changes now.

File: docs/content/services/networking/application-gateway/_index.md [hunk 5]
```diff
@@ -227,11 +231,29 @@ Plan for backend maintenance by using connection draining. Connection draining h
 - [Application Gateway Connection Draining HTTP Settings](https://learn.microsoft.com/azure/application-gateway/configuration-http-settings#connection-draining)
 
 **Resource Graph Query/Scripts**
-
```
[the remark this answers is not in the document]
Comment (jimays-avila) [reply docs/content/services/networking/application-gateway/_index.md:230]: I will review this.

**Review by jimays-avila — COMMENTED** [review]
Are there any other changes that need to be made

**Review by oZakari — APPROVED** [review]

**Pull request merged**